### PR TITLE
fix: use case-insensitive comparison for GitHub orgs to fix RBAC #2499

### LINKED ISSUE #2499: Org definition using incorrect case can silently fail to pull github groups.
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/dexidp/dex/blob/master/.github/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/dexidp/dex/issues) for an issue that matches the one I want to file, without success.
- [X] I am not looking for support or already pursued the available [support channels](https://github.com/dexidp/dex/issues/new/choose) without success.

### Version

v2.30.0-dirty

### Storage Type

Kubernetes

### Installation Type

Custom Helm chart

### Expected Behavior

A warning would be logged if the defined org within the github connector does not match the case used in the login name (as returned by the github api).

### Actual Behavior

Defining `A1Example` in the config when the org name is `a1example` will allow successful authentication but with an empty group list, and subsequently fail to apply any group specific access policies.

### Steps To Reproduce

_No response_

### Additional Information

This was discovered using dex bundled within the following helm chart:
https://artifacthub.io/packages/helm/argo/argo-cd/3.29.5#dex

### Configuration

```yaml
argo-cd:
  server:
    configEnabled: true
    config:
      url: https://argo.a1example.com
      dex.config: |-
        connectors:
        - type: github
          id: github
          name: GitHub
          config:
            clientID: $GITHUB_CLIENT_ID
            clientSecret: $GITHUB_CLIENT_SECRET
            orgs:
            - name: A1Example
```

### Logs

```shell
Correct case:
`time="2022-04-26T15:02:55Z" level=info msg="login successful: connector \"github\", username=\"Fil Ivovic\", preferred_username=\"fivovic\", email=\"f.ivovic@a1example.co.uk\", groups=[\"a1example:admins\"]"`

Incorrect case:
`time="2022-04-26T15:02:55Z" level=info msg="login successful: connector \"github\", username=\"Fil Ivovic\", preferred_username=\"fivovic\", email=\"f.ivovic@a1example.co.uk\", groups=[]"`
```

### PR DESCRIPTION
**Pull request by venkatamutyala**
#### Overview
Previously, configuring `orgs: [{name: "MyOrg"}]` when the actual org was `myorg` would silently return empty groups, causing users to be denied access despite being valid org members

#### What this PR does / why we need it
adds case-insensitive comparison for GitHub organization names

Closes #2499

#### Special notes for your reviewer

For testing, I did a docker build, updated my dex install to use this custom image, i changed my github org to have mixed casing and then tested the login via dex github connector and had no issues.

### CODE REVIEW

File: connector/github/github.go [hunk 1]
```diff
@@ -706,7 +706,7 @@ func (c *githubConnector) teamsForOrg(ctx context.Context, client *http.Client,
 		}
 
 		for _, t := range teams {
-			if t.Org.Login == orgName {
+			if strings.EqualFold(t.Org.Login, orgName) {
```
Comment (nabokihms) [inline connector/github/github.go:715]: Seems good to me, but let's add the comment on why the comparison here is case-insensitive. 

Also, it's kind of a breaking change for people who, for some reason, used an org with an upper case in their configs. Users from these orgs can log in now, but they could not before this change.

**Review by nabokihms — CHANGES_REQUESTED** [review]
I have one comment and a question. The original issue was about the org name, but this PR fixes team names, not the org name. Am I missing something?

**Comment by venkatamutyala** [comment]
> I have one comment and a question. The original issue was about the org name, but this PR fixes team names, not the org name. Am I missing something?

@nabokihms  I think the function is a little misleading. It's fetching the users teams across all orgs and then filters them by org name. It's the comparison on org name where it's silently failing because it's case sensitive. I updated the link to the API call as well in my most recent commit

File: connector/github/github.go (hunk 1 above)
Comment (venkatamutyala) [reply connector/github/github.go:715]: @nabokihms Added a comment and updated the link as well to go straight to the relevant API call.

RE: breaking change. 

Potentially. I would think if you didn't want an org to have access you would have removed the org entirely rather than relying on the current case mismatch as security.

Comment (venkatamutyala) [reply connector/github/github.go:715]: @nabokihms thoughts?

Comment (nabokihms) [reply connector/github/github.go:715]: Nit: let's make the comment a bit more specific:

```
// GitHub org names are case-insensitive; the API returns canonical (lowercase) form.
// [link-to-the-API-doc]
```

File: connector/github/github.go [hunk 2]
```diff
@@ -706,7 +706,8 @@ func (c *githubConnector) teamsForOrg(ctx context.Context, client *http.Client,
 		}
 
 		for _, t := range teams {
-			if t.Org.Login == orgName {
+			// case-insensitive check to avoid silent failures when config casing is different than GitHub
+			if strings.EqualFold(t.Org.Login, orgName) {
```
Comment (nabokihms) [inline connector/github/github.go:715]: Could you also add tests for this edge-case?

Comment (nabokihms) [inline connector/github/github.go:715]: The warning is required if there is a case mismatch.
1. In the final groups, the ID token org name will be in lowercase, so we need to warn users that it was converted to lowercase
2. we need to suggest using lowercase orgs in the config 

The warning will solve both.

**Review by nabokihms — CHANGES_REQUESTED** [review]
Some suggestions

File: connector/github/github.go (hunk 1 above)
Comment (venkatamutyala) [reply connector/github/github.go:715]: Done. @nabokihms

File: connector/github/github.go (hunk 2 above)
Comment (venkatamutyala) [reply connector/github/github.go:715]: @nabokihms Added. Please let me know if you have any feedback on it.

Comment (venkatamutyala) [reply connector/github/github.go:715]: Added. Let me know if you have any feedback on the wording.

**Comment by venkatamutyala** [comment]
@nabokihms sorry for the delay. Been a busy month. Just made the requested changes. Please take another look when you have a moment.

**Comment by venkatamutyala** [comment]
@nabokihms 🙏🏾

**Comment by venkatamutyala** [comment]
Hey @nabokihms any chance you have time to give this a review?
